apiserver/facades/client/spaces/opfactory.go

```diff
@@ -49,9 +55,19 @@ func (f *opFactory) NewRemoveSpaceModelOp(fromName string) (state.ModelOperation
 // for renaming a space.
 func (f *opFactory) NewRenameSpaceModelOp(fromName, toName string) (state.ModelOperation, error) {
 	space, err := f.st.SpaceByName(fromName)
-	controllerSettings := f.st.NewControllerSettings()
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return NewRenameSpaceModelOp(f.st.IsController(), controllerSettings, &renameSpaceState{f.st}, space, toName), nil
+	return NewRenameSpaceModelOp(
+		f.st.IsController(), f.st.NewControllerSettings(), &renameSpaceState{f.st}, space, toName), nil
+}
+
+// NewMoveSubnetsOp (OpFactory) returns an operation
+// to move a list of subnets to a space.
+func (f *opFactory) NewMoveSubnetsOp(spaceName string, subnets []MovingSubnet) (MoveSubnetsOp, error) {
+	space, err := f.st.SpaceByName(spaceName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return NewMoveSubnetsOp(networkingcommon.NewSpaceShim(space), subnets), nil
```

Comment:
I don't like the way this factory holds together. We have no test coverage of the factory itself, which accepts a raw `*state.State`. I anticipate a follow-up to break `Backing` into more localised interfaces, eventually removing the reliance on `networking/common`.

Reply:
Agreed, the factory isn't pretty here.